[inductor] Fix CudaStreamGuard in AOTInductor ABI compatible mode

desertfire opened the pull request:
Summary: Use a RAII class to wrap around at::cuda::CUDAStreamGuard. Previous implementation didn't follow the exact CUDAStreamGuard behavior.

Test Plan: CI

Differential Revision: D49355542

chenyang78 reviewed (approved):
LGTM. 

Wondering why the old approach failed? Thanks!

desertfire committed:
[inductor] Fix CudaStreamGuard in AOTInductor ABI compatible mode (#109471)

Summary:

Use a RAII class to wrap around at::cuda::CUDAStreamGuard. Previous implementation didn't follow the exact CUDAStreamGuard behavior.

Test Plan: CI

Reviewed By: frank-wei, chenyang78

Differential Revision: D49355542
